### ISSUE TITLE
[Data] Include image class id in the returned datasets of `ray.data.read_images()`.

### ISSUE THREAD

**Issue by woshiyyya** [ray-project/ray#32989]
### Description

It is a common use case to read data from a image folders, where the folder names are the class names.

```python
partitioning = Partitioning("dir", field_names=["class"], base_dir=data_folder)
ds = ray.data.read_images(data_folder, size=(256, 256), partitioning=partitioning, mode="RGB")
```
However, the `ray.data.read_images` api returns a datasets with two columns ["image", "class"]. Actually we need the integer class id more than the class string name. Otherwise we have to write an additional `ds.map_batches()` to map class names to integers.

For example, when we iterate throught a torchvision `ImageFolder`, it will generate two tensors, one with image and another with label_id.

```python
dataset = datasets.ImageFolder(data_folder, data_transforms)
dataloader = DataLoader(dataset, batch_size=3)
print(next(iter(dataloader)))
>>>       
[tensor([[[[ 0.0912,  0.0398, -0.0287,  ...,  1.0673,  1.0844,  1.0502],
          [ 0.3823,  0.3481,  0.3138,  ...,  1.0331,  1.1187,  1.1015],
          [ 0.6392,  0.6221,  0.5707,  ...,  1.0159,  1.0844,  1.1187],
    ...,
          [-0.3055, -0.3927, -0.4275,  ..., -0.7238, -0.7064, -0.8633],
          [-0.3230, -0.3927, -0.4450,  ..., -0.7761, -0.8981, -0.9330],
          [-0.3753, -0.3753, -0.4275,  ..., -0.8284, -1.0027, -0.9504]]]]), tensor([0, 0])]
```

**Comment by cszhu** [comment]
This P2 issue has seen no activity in the past 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please comment and remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
